class-tgm-plugin-activation.php

```diff
@@ -249,7 +249,7 @@ public function init() {
 			// After this point, the plugins should be registered and the configuration set.
 
 			// Proceed only if we have plugins to handle.
-			if ( $this->plugins ) {
+			if ( ! empty( $this->plugins ) ) {
 				$sorted = array();
```

Comment:
Where do I start ? Empty is hardly *ever* the correct function to use.

1. empty does not test for type while you need an array two lines later for the `foreach` which will throw a notice if it doesn't get one
2. empty will pass on a non-empty string, boolean true, a non-zero integer and float etc
3. empty actually doesn't pass on the string '0' (not relevant in this case, but confuses people often)
etc...

Always test specifically for what you want to know - here: do we have a non-empty array. And in this case neither `if ( $this->plugins )` nor `if ( ! empty( $this->plugins ) )` does.

See: http://phpcheatsheets.com/test/